bleak/backends/bluezdbus/client.py

```diff
@@ -104,109 +135,289 @@ async def connect(self, **kwargs) -> bool:
+                            defs.GATT_CHARACTERISTIC_INTERFACE
+                        ]
+                        uuid = characteristic["UUID"].value
+                        handle = extract_service_handle_from_path(obj["Characteristic"])
+                        self.services.add_descriptor(
+                            BleakGATTDescriptorBlueZDBus(obj, path, uuid, handle)
+                        )
+
+                try:
+                    reply = await asyncio.wait_for(
+                        self._bus.call(
+                            Message(
+                                destination=defs.BLUEZ_SERVICE,
+                                interface=defs.DEVICE_INTERFACE,
+                                path=self._device_path,
+                                member="Connect",
+                            )
+                        ),
+                        timeout,
+                    )
+                    assert_reply(reply)
+                except BaseException:
+                    # calling Disconnect cancels any pending connect request
+                    try:
+                        reply = await self._bus.call(
+                            Message(
+                                destination=defs.BLUEZ_SERVICE,
+                                interface=defs.DEVICE_INTERFACE,
+                                path=self._device_path,
+                                member="Disconnect",
+                            )
+                        )
+                        try:
+                            assert_reply(reply)
+                        except BleakDBusError as e:
+                            # if the object no longer exists, then we know we
+                            # are disconnected for sure, so don't need to log a
+                            # warning about it
+                            if e.dbus_error != ErrorType.UNKNOWN_OBJECT.value:
+                                raise
+                    except Exception as e:
+                        logger.warning(
+                            f"Failed to cancel connection ({self._device_path}): {e}"
+                        )
+
+                    raise
+
+            if self.is_connected:
+                logger.debug(f"Connection successful ({self._device_path})")
             else:
-                raise BleakError(str(e))
+                raise BleakError(
+                    f"Connection was not successful! ({self._device_path})"
+                )
 
-        if await self.is_connected():
-            logger.debug("Connection successful.")
-        else:
-            await self._cleanup_all()
-            raise BleakError(
-                "Connection to {0} was not successful!".format(self.address)
-            )
+            # Create a task that runs until the device is disconnected.
+            self._disconnect_monitor_event = asyncio.Event()
+            asyncio.create_task(self._disconnect_monitor())
```

Comment:
`asyncio.create_task` is [Python 3.7+](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task). We should use `asyncio.ensure_future` for now instead.

Ran into this using system python (3.6.7) in my Ubuntu 18.04.3 LTS dev machine.

Reply:
Just now found out that all my new async callbacks in the `develop` branch also uses `create_task`. Will remedy that before next release...

Reply:
I went ahead and fixed all instances in this PR

Reply:
👍🏻 I did a commit for it on #450 as well.